src/services/goToDefinition.ts

```diff
@@ -26,10 +26,22 @@ namespace ts.GoToDefinition {
             return label ? [createDefinitionInfoFromName(typeChecker, label, ScriptElementKind.label, node.text, /*containerName*/ undefined!)] : undefined; // TODO: GH#18217
         }
 
-        if (node.kind === SyntaxKind.ReturnKeyword) {
-            const functionDeclaration = findAncestor(node.parent, n =>
-                isClassStaticBlockDeclaration(n) ? "quit" : isFunctionLikeDeclaration(n)) as FunctionLikeDeclaration | undefined;
-            return functionDeclaration ? [createDefinitionFromSignatureDeclaration(typeChecker, functionDeclaration)] : undefined;
+        switch (node.kind) {
+            case SyntaxKind.ReturnKeyword:
+                const functionDeclaration = findAncestor(node.parent, n =>
+                  isClassStaticBlockDeclaration(n) ? "quit" : isFunctionLikeDeclaration(n)) as FunctionLikeDeclaration | undefined;
+                return functionDeclaration ? [createDefinitionFromSignatureDeclaration(typeChecker, functionDeclaration)] : undefined;
+            case SyntaxKind.DefaultKeyword:
+                if (!isDefaultClause(node.parent)) {
+                    break;
+                }
+                // falls through
+            case SyntaxKind.CaseKeyword:
+                const switchStatement = findAncestor(node.parent, isSwitchStatement);
+                if (switchStatement) {
+                    return [createDefinitionInfoFromNode(switchStatement, "switch")];

```

Comment:
This is setting definition as whole switch block. want only keyword switch i think or switch expression ?

Reply:
Sorry, that was my suggestion.